addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/handler/OpenDaikinAcUnitHandler.java

```diff
@@ -0,0 +1,267 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void pollStatus() throws IOException {
+        ControlInfo controlInfo = webTargets.getControlInfo();
+        updateStatus(ThingStatus.ONLINE);
+        if (controlInfo != null) {
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
+            if (controlInfo.temp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp.get());
+            }
+
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_DIR, new StringType(controlInfo.fanMovement.name()));
+        }
+
+        SensorInfo sensorInfo = webTargets.getSensorInfo();
+        if (sensorInfo != null) {
+            if (sensorInfo.indoortemp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp.get());
+            } else {
+                updateState(OpenDaikinBindingConstants.CHANNEL_INDOOR_TEMP, UnDefType.UNDEF);
+            }
+
+            if (sensorInfo.outdoortemp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp.get());
+            } else {
+                updateState(OpenDaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, UnDefType.UNDEF);
+            }
+
+            if (sensorInfo.indoorhumidity.isPresent()) {
+                updateState(OpenDaikinBindingConstants.CHANNEL_HUMIDITY,
+                        new DecimalType(sensorInfo.indoorhumidity.get()));
+            } else {
+                updateState(OpenDaikinBindingConstants.CHANNEL_HUMIDITY, UnDefType.UNDEF);
+            }
+        }
+    }
+
+    private void updateTempState(String channel, double temp) {
+        QuantityType<Temperature> tempToUpdateWith = new QuantityType<Temperature>(new DecimalType(temp),
+                SIUnits.CELSIUS);
+
+        updateState(channel, tempToUpdateWith);
+    }
+
+    private void changePower(boolean power) throws OpenDaikinCommunicationException {
+        ControlInfo info = webTargets.getControlInfo();
+        info.power = power;
+        webTargets.setControlInfo(info);
+    }
+
+    private void changeSetPoint(double temp) throws OpenDaikinCommunicationException {
+        ControlInfo info = webTargets.getControlInfo();
+
+        if (useFahrenheitForChannel(OpenDaikinBindingConstants.CHANNEL_AC_TEMP)) {
+            info.temp = Optional.of(fToC(temp));
+        } else {
+            info.temp = Optional.of(temp);
+        }
+
+        webTargets.setControlInfo(info);
+    }
+
+    private boolean useFahrenheitForChannel(String channel) {
```

Comment:
It seems that you already use UoM, it seems however that you need to remove/trim some old methods.